### PR TITLE
mget with an alias shouldn't ignore alias routing

### LINKED ISSUE #25696: _mget with an alias using routing doesn't automatically use the routing value
**Elasticsearch version**: 2.4.5

**Description of the problem including expected versus actual behavior**: In all versions 2.4.2+ `_mget` used for an alias that has routing defined will not use the routing value to get the document and will report back an error:

```
            "type": "illegal_argument_exception",
            "reason": "routing is required for [test]/[test]/[1]"
```

**Steps to reproduce**:

```
DELETE test

PUT test
{
  "mappings": {
    "test": {
      "_routing": {
        "required": true
      }
    }
  }
}

POST test/test/_bulk
{"index":{"_routing":123,"_id":1}}
{"some_field":"bla"}
{"index":{"_routing":123,"_id":2}}
{"some_field":"bla"}

POST /_aliases
{
  "actions": [
    {
      "add": {
        "index": "test",
        "alias": "test_alias",
        "routing": "123"
      }
    }
  ]
}

GET /_mget
{
  "docs": [
    {
      "_index": "test_alias",
      "_type": "test",
      "_routing": "123",
      "_id": "1" 
    }
  ]
}

GET /_mget
{
  "docs": [
    {
      "_index": "test_alias",
      "_type": "test",
      "_id": "1" 
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by imotov**
Closes #25696

### CODE REVIEW

**Comment by imotov** [comment]
@javanna I think this should fix it. Could you take a look?

File: core/src/test/java/org/elasticsearch/mget/SimpleMgetIT.java [hunk 1]
```diff
@@ -106,6 +106,23 @@ public void testThatMgetShouldWorkWithMultiIndexAlias() throws IOException {
...
+        client().prepareIndex("alias1", "test", "1").setSource(jsonBuilder().startObject().field("foo", "bar").endObject())
+            .setRefreshPolicy(IMMEDIATE).get();
+
+        MultiGetResponse mgetResponse = client().prepareMultiGet()
+            .add(new MultiGetRequest.Item("alias1", "test", "1"))
+            .get();
+        assertThat(mgetResponse.getResponses().length, is(1));
+
+        assertThat(mgetResponse.getResponses()[0].getIndex(), is("test"));
+        assertThat(mgetResponse.getResponses()[0].isFailed(), is(false));
```
Comment (javanna) [inline core/src/test/java/org/elasticsearch/mget/SimpleMgetIT.java]: assertEquals and assertFalse?

**Review by javanna — APPROVED** [review]
LGTM thanks @imotov

**Comment by javanna** [comment]
We may want to backport to 5.5?

**Comment by javanna** [comment]
thanks a lot for fixing this @imotov !!!

**Pull request merged**